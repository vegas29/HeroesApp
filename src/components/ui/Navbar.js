import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    return (
        <nav className="bg-blue-500 p-5 shadow-lg text-white text-xl">
            
            <div className="container mx-auto flex flex-col md:flex-row items-center  gap-5 ">
                <Link 
                    className="navbar-brand" 
                    to="/"
                >
                    Asociaciones
                </Link>

                <div className="flex flex-col md:flex-row gap-5">
                    <NavLink 
                        className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg text-center" 
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg text-center" 
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>
                </div>

                <div className="flex md:justify-end md:w-full ">
                    <ul className="">
                        <NavLink 
                            activeClassName="active"
                            className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg" 
                            exact
                            to="/login"
                        >
                            <i class="fa-solid fa-right-from-bracket"></i>
                        </NavLink>
                    </ul>
                </div>
            </div>
            
        </nav>
    )
}