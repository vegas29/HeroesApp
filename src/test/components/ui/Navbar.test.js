import { mount } from "enzyme";
import { AuthContext } from "../../../auth/authContext";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

const mockNavigate= jest.fn();

jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}))

const contextValue = {
    user:{
        name: 'Pedro',
        logged:true
    },
    dispatch:jest.fn()
}

const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/']}>
            <Routes>
                <Route path="/" element={<Navbar/>}/>
            </Routes>
        </MemoryRouter>
    </AuthContext.Provider>
)


describe('Pruebas en <Navbar/>', ()=>{
    test('Debe de mostrar correctamente', ()=>{
        
        

        //Snapshopt
        expect(wrapper).toMatchSnapshot();
        console.log(wrapper.html());

        //span.font-semibold == pedro
        expect(wrapper.find('span.font-semibold').text().trim()).toBe('Bienvenido Pedro')
    })

    test('Debe de llamar el logout, llamar el navigate y el dispatch con los argumentos', ()=>{
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith({"type": types.logout});
        expect(mockNavigate).toHaveBeenCalledWith('/login', {replace:true});
    })
})