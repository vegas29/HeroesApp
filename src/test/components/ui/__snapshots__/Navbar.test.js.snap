// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar/> Debe de mostrar correctamente 1`] = `
<nav
  className="bg-blue-500 p-5 shadow-lg text-white text-xl"
>
  <div
    className="container mx-auto flex flex-col md:flex-row items-center  gap-5 "
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="flex flex-col md:flex-row gap-5 items-center"
    >
      <a
        className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg text-center "
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg text-center "
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg text-center "
        href="/search"
        onClick={[Function]}
      >
        Buscar
      </a>
    </div>
    <div
      className="flex gap-2 md:justify-end md:w-full"
    >
      <ul
        className="flex items-center gap-5"
      >
        <span
          className="font-semibold"
        >
          Bienvenido 
          Pedro
        </span>
        <button
          className="hover:bg-blue-600 py-2 px-3 hover:px-3 hover:py-2 hover:rounded-lg"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-right-from-bracket"
          />
        </button>
      </ul>
    </div>
  </div>
</nav>
`;
